src/interfaces/interfaces.ts

```diff
@@ -0,0 +1,11 @@
+export interface Subject {
+    attach(observer: any): void;
+    detach(observer: any): void;
+    notify(): void;
+}
+
+export interface Rendereble {
+    _selector: string;
+    render(...args: any[]): void;
```

Comment:
I want to suggest you make the following improvements to make typings better and follow OOP concepts. 
Let's rename `Renderable` to `BaseView`/`View` and create separate interfaces for components view to make it possible to use these interfaces for custom views. 

```typescript
export interface BaseView {
    selector: string;
    render(...args : any[]): void; // Could have better type
}

export interface TableView<ReportItem> extends BaseView {
    render(columns: ReportColumns, bodyModel: ReportItem[], sortModel: SortModel): void;
}

export interface FilterView extends BaseView {
    render(filters: Filter[]): void;
}

export interface PaginationView {
    render(paginationParams: PaginationParams): void;
}

export class DefaultTableView<ReportItem> implements TableView<ReportItem> {
    // implementation
}

export class DefaultFilterView implements FilterView {
    // implementation
}

export class DefaultPaginationView implements PaginationView {
    // implementation
}
```